DataStructureClasses/addContact.ts

```diff
@@ -0,0 +1,50 @@
+import ContactNotFoundError from "./error";
+
+
+export default class PhoneBook {
+    contacts: any[];
+    constructor() {
+        this.contacts = [];
+    }
+    addContact(name: string, phoneNumber: string ,alterPhonenumber:string) {
+        try{
+            this.contacts.push({ name, phoneNumber ,alterPhonenumber});
+            console.log(`Contact "${name}" added successfully.`);
+        }catch(error){
+            throw new ContactNotFoundError(error.message)
+
+        }
+    }
+    updateContact(name: string, newPhoneNumber: string, alterNatenewPhoneNumberToAdd: string) {
+        try{
+        const contact = this.findContact(name);
+        if (contact) {
+            contact.phoneNumber = newPhoneNumber;
+            console.log(`Contact "${name}" updated successfully.`);
+        } else {
+            console.log(`Contact "${name}" not found.`);
+        }
+      }catch(error){
```

Comment:
where are you updating the contact name or alternate number. You have a parameter **alterNatenewPhoneNumberToAdd** but you are not using it anywhere

Reply:
resolved